src/model/document.ts

```diff
@@ -130,9 +131,9 @@ export class Document implements Feature, Queryable {
     this.languageAnalysis = languageAnalysis;
 
     if (!base.isInline) {
-      this.identifiers.add(this.url);
+      (this.identifiers as Set<string>).add(this.url);
```

Comment:
could we make a function that turns all of our immutable types into mutable types?